### PR TITLE
Refactor: general update of Exception Mapping

### PR DESCRIPTION
**Pull request by BenitoVisone**
This **PR** introduces a general refactoring for Exception Mapping in order to return correct error type based on route's content type.

This changes involved a general review of the Exception Mapping system.

### CODE REVIEW

File: src/oneid/oneid-common/model/src/main/java/it/pagopa/oneid/common/model/exception/enums/ErrorCode.java [hunk 1]
```diff
@@ -6,18 +6,27 @@
...
   GENERIC_AUTHN_REQUEST_CREATION("GENERIC_AUTHN_REQUEST_CREATION",
       "Generic AuthnRequest Exception"),
-  SAMLUTILS_ERROR("SAMLUTILS_ERROR", "Generic exception inside SAMLUtils"),
-  CALLBACK_URINOT_FOUND("CALLBACK_URINOT_FOUND", ""),
-  CLIENT_NOT_FOUND("CLIENT_NOT_FOUND", ""),
-  IDPNOT_FOUND("IDPNOT_FOUND", ""),
+  SAML_UTILS_ERROR("SAML_UTILS_ERROR", "Generic exception inside SAMLUtils"),
+  CALLBACK_URI_NOT_FOUND("CALLBACK_URI_NOT_FOUND",
+      "Callback URI not found from in Client metadata"),
+  CLIENT_NOT_FOUND("CLIENT_NOT_FOUND", "Client not registeres"),
```
Comment (GiuMontesano) [inline src/oneid/oneid-common/model/src/main/java/it/pagopa/oneid/common/model/exception/enums/ErrorCode.java:15]: ~"Client no registeres"~
"Client no registered"

Comment (BenitoVisone) [inline src/oneid/oneid-common/model/src/main/java/it/pagopa/oneid/common/model/exception/enums/ErrorCode.java:15]:
```suggestion
  CLIENT_NOT_FOUND("CLIENT_NOT_FOUND", "Client not registered"),
```

**Review by GiuMontesano — APPROVED** [review]
LGTM

**Review by giuseppe-gangemi — APPROVED** [review]
lgtm

**Pull request merged**